ext/ddtrace_profiling_native_extension/heap_recorder.c

```diff
@@ -1,23 +1,129 @@
 #include "heap_recorder.h"
 #include <pthread.h>
 #include "ruby/st.h"
-#include "ruby/util.h"
 #include "ruby_helpers.h"
 #include <errno.h>
 
+#define MAX_FRAMES_LIMIT 10000
+
+// A compact representation of a stacktrace frame for a heap allocation.
+typedef struct {
+  char *name;
+  char *filename;
+  int64_t line;
```

Comment:
I had initially kept the typing from the ddog_prof_Location struct. 

I think what you say makes sense but going with a `int32_t` since even the ruby native methods we're using seem to operate on signed `ints` (despite seeming to use 0 as invalid/unknown markers). I don't think having 2M vs 4M lines of code in a Ruby source file makes enough of a difference anyway?

Reply:
Yeah, that seems fine!